git_theta/lsh/pool.py

```diff
@@ -0,0 +1,64 @@
+"""Class for deterministically supplying pre-computed random values"""
+
+import numpy as np
+import numba as nb
+from numba import int32, float32
+import sys
+
+# TODO(bdlester): importlib.resources doesn't have the `.files` API until python
+# version `3.9` so use the backport even if using a python version that has
+# `importlib.resources`.
+if sys.version_info < (3, 9):
+    import importlib_resources
+else:
+    import importlib.resources as importlib_resources
+
+
+# RandomnessPool is a singleton class so we don't read pool data off disk multiple times per git-theta command
+SINGLETON = None
+
+spec = [("pool", nb.float64[:]), ("index_hashes", nb.int64[:])]
```

Comment:
Is there a reason to work in `int64` over `uint64`? Generally I think people work with uints when they are going to be doing bitwise things.

Reply:
From some testing (and you mentioned this below), `np.mod` of an unsigned type seems to return a float. To avoid this just used int64 everywhere.